### PR TITLE
fix: bump snowflake sdk to 14

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash-internal-work/issues/2098

test-cli
test-frontend
test-backend

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts [hunk 1]
```diff
@@ -193,7 +194,7 @@ export class SnowflakeWarehouseClient extends WarehouseBaseClient<CreateSnowflak
         let connection: Connection;
         try {
             connection = createConnection(this.connectionOptions);
-            await Util.promisify(connection.connect)();
+            await Util.promisify(connection.connect.bind(connection))();
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts:197]: this should fix the `this...` is undefined, thing, right ?

File: packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts [hunk 2]
```diff
@@ -178,7 +179,7 @@ export class SnowflakeWarehouseClient extends WarehouseBaseClient<CreateSnowflak
             ...(credentials.accessUrl?.length
                 ? { accessUrl: credentials.accessUrl }
                 : {}),
-        } as ConnectionOptions; // force type because accessUrl property is not recognised
+        };
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts:182]: no need to force type anymore ?

**Review by rephus — APPROVED** [review]
I don't understand exactly what is the fix, but I assume it is some obscure detail in the library...

File: packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts:197]: yes!

File: packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts (hunk 2 above)
Comment (IrakliJani) [reply packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts:182]: nope, it does not complain anymore

**Comment by IrakliJani** [comment]
waiting for all the tests to pass

File: packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts:197]: check this discussion for the context: https://github.com/snowflakedb/snowflake-connector-nodejs/issues/929#issuecomment-2393283248

**Comment by IrakliJani** [comment]
unrelated test failures. merging

**Pull request merged**